### PR TITLE
[SPARK-47442][CORE][TEST] Use port 0 to start worker servers in MasterSuite

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

This PR proposes to use port 0 to start the worker server.

### Why are the changes needed?

More stable test by:

* Avoids the port conflicts between workers
* Avoids the port conflicts with local service

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manually tested.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
Thank you, @Ngone51 and @HyukjinKwon !

+1, LGTM.

File: core/src/test/scala/org/apache/spark/deploy/master/MasterSuiteBase.scala [hunk 1]
```diff
@@ -49,7 +49,7 @@ object MockWorker {
 class MockWorker(master: RpcEndpointRef, conf: SparkConf = new SparkConf) extends RpcEndpoint {
   val seq = MockWorker.counter.incrementAndGet()
   val id = seq.toString
-  override val rpcEnv: RpcEnv = RpcEnv.create("worker", "localhost", seq,
+  override val rpcEnv: RpcEnv = RpcEnv.create("worker", "localhost", 0,
```
Comment (cloud-fan) [inline core/src/test/scala/org/apache/spark/deploy/master/MasterSuiteBase.scala:52]: just for my education, port 0 means using any available port?

Comment (cloud-fan) [reply core/src/test/scala/org/apache/spark/deploy/master/MasterSuiteBase.scala:52]: if so can we add a comment to mention it?

Comment (Ngone51) [inline core/src/test/scala/org/apache/spark/deploy/master/MasterSuiteBase.scala:52]:
```suggestion
  // Use port 0 to start the server with a random free port
  override val rpcEnv: RpcEnv = RpcEnv.create("worker", "localhost", 0,
```

Comment (Ngone51) [reply core/src/test/scala/org/apache/spark/deploy/master/MasterSuiteBase.scala:53]: > if so can we add a comment to mention it?

Added.

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 4.0.0. Thank you, @Ngone51 and all.

**Pull request closed without merging**